### PR TITLE
DAOS-14075 bio: Sync LED state on engine start

### PR DESCRIPTION
**Pull request by tanabarr**
SSDs can store LED state between server restarts so load state for
each bdevs after they have all been created. Additionally, don't
require state receiver in bio_led_manage.

Includes some small bio_ut fixes to enable logging in test and run 
in a local environment with linux file written to /dev/vdb.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by kjacque — APPROVED** [review]
Looks good to me, with the caveat that I don't have deep knowledge of the SPDK APIs.

File: src/bio/bio_xstream.c [hunk 1]
```diff
@@ -972,9 +973,32 @@ init_bio_bdevs(struct bio_xs_context *ctxt)
...
-			break;
+			return rc;
 	}
-	return rc;
+
+	for (bdev = spdk_bdev_first(); bdev != NULL; bdev = spdk_bdev_next(bdev)) {
+		if (nvme_glb.bd_bdev_class != get_bdev_type(bdev))
+			continue;
+
+		d_bdev = lookup_dev_by_name(spdk_bdev_get_name(bdev));
```
Comment (NiuYawei) [inline src/bio/bio_xstream.c:983]: why not directly call this lookup_dev_by_name() & bio_led_mange() in above loop?

**Review by NiuYawei — APPROVED** [review]

**Review by wangshilong — DISMISSED** [review]

File: src/bio/bio_xstream.c (hunk 1 above)
Comment (tanabarr) [reply src/bio/bio_xstream.c:983]: The led reset called through bio_led_manage() fails when checking all the devices for faulty state if the blobstores are not created for all bdevs. check_faulty() calls bio_dev_list() which requires the above loop to have completed. Make sense?

Comment (NiuYawei) [reply src/bio/bio_xstream.c:983]: I see, thanks.

**Comment by tanabarr** [comment]
@jolivier23 think there may be a regression made when changing to utest in the bio_ut flow. Looks  like the segfault in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12749/13/pipeline/ maybe caused by insufficient size of `/dev/vdb` used as the AIO file in the tests. Please check if my fix (adding `conv=notrunc` to dd call in utils/run_utest.py makes sense. Thanks @NiuYawei for helping to debug.

**Comment by tanabarr** [comment]
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12749/14/pipeline/614/ proves it's not a regression related to utest. https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12749/13/artifact/unit_test_bdev_logs/bin-vos_tests--A-50_2/daos.log/*view*/ shows it's an issue in this PR.

File: src/bio/bio_xstream.c [hunk 2]
```diff
@@ -972,9 +973,32 @@ init_bio_bdevs(struct bio_xs_context *ctxt)
...
+		if (d_bdev == NULL) {
+			D_ERROR("Device %s doesn't exist\n", spdk_bdev_get_name(bdev));
```
Comment (knard-intel) [inline src/bio/bio_xstream.c:985]: Nit, if you have to merge and repush `DL_ERROR` could be better suited ?

**Review by knard-intel — APPROVED** [review]
LGTM

**Review by knard-intel — DISMISSED** [review]

**Review by wangshilong — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]

**Comment by tanabarr** [comment]
The consistent known issue that appears in functional VM tests when running with Features: control pragma is DAOS-14169 (DmgPoolQueryRanks).

**Review by wangshilong — APPROVED** [review]

**Comment by mchaarawi** [comment]
@NiuYawei could you please provide a review for this PR? tia

**Comment by mchaarawi** [comment]
> @NiuYawei could you please provide a review for this PR? tia

nevermind, i see you already approved before.

**Pull request merged**